src/cpu/aarch64/acl_inner_product_utils.cpp

```diff
@@ -0,0 +1,158 @@
+/*******************************************************************************
+* Copyright 2021 Arm Ltd. and affiliates
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "cpu/aarch64/acl_inner_product_utils.hpp"
+
+namespace dnnl {
+namespace impl {
+namespace cpu {
+namespace aarch64 {
+
+namespace acl_inner_product_utils {
+
+using namespace format_tag;
+using namespace utils;
+using namespace status;
+
+status_t init_conf_ip(acl_ip_conf_t &aip, memory_desc_t &src_md,
+        memory_desc_t &wei_md, memory_desc_t &dst_md, memory_desc_t &bias_md,
+        const inner_product_desc_t &ipd, const primitive_attr_t &attr) {
+    const memory_desc_wrapper src_d(&src_md);
+    const memory_desc_wrapper wei_d(&wei_md);
+    const memory_desc_wrapper dst_d(&dst_md);
+    const memory_desc_wrapper bia_d(&bias_md);
+
+    // Compute Library currently supports forward propagation only
+    const prop_kind_t prop_kind = ipd.prop_kind;
+    const bool is_fwd = (prop_kind == dnnl_forward_training)
+            || (prop_kind == dnnl_forward_inference);
+    if (!is_fwd) return status::unimplemented;
+
+    const int with_groups = wei_d.ndims() == src_d.ndims() + 1;
+    const int ndims = src_d.ndims();
+
+    // There are two sub-cases: src & wei tensors are either 2- or 4-dimensional
+    const bool is_2d = (ndims == 2) && (wei_d.ndims() == 2);
```

Comment:
>4d tensors with one of dims equal to 1

But `ndims` in such tensors will still be 4, right?

Reply:
Correct, but it can be done internally so user doesn't have to do it themselves. For example, inside oneDNN in gemm-based inner product we interpret all 3d+ descriptors as 2d ones in case they are dense. Here is an example:
* gemm_inner_product [checks](https://github.com/oneapi-src/oneDNN/blob/8f40cd636b2ad278f6840dfbaa98ee6ff8405b17/src/cpu/gemm_inner_product.hpp#L55) that mds are compatible using `dense_gemm_consitency_check`
* gemm_inner_product [executes 2d gemm](https://github.com/oneapi-src/oneDNN/blob/8f40cd636b2ad278f6840dfbaa98ee6ff8405b17/src/cpu/gemm_inner_product.cpp#L55) on 2d matrixes with sizes [batch, channels * depth * height * width]

This optimization can be implemented later, because reference code will cover 3d case. The optimization is pretty simple, but extends coverage significantly.

Reply:
@igorsafo, thanks for explaining, now this point is clear. I tested only 2d and 4d use cases so far and suggest to add 3d support later as a separate commit.